msresist/expectation_maximization.py

```diff
@@ -58,13 +63,13 @@ def EM_clustering(data, info, ncl, seqDist=None, gmmIn=None):
                 nDist[-1].summarize(d[:, -1], weights=weights)
                 nDist[-1].from_summaries()
 
-                dists.append(IndependentComponentsDistribution(nDist))
+                dists.append(IndependentComponentsDistribution(nDist, weights=seqWarr))
 
             gmm = GeneralMixtureModel(dists)
         else:
             gmm = gmmIn
 
-        gmm.fit(d, max_iterations=200, verbose=False, stop_threshold=1e-6)
+        gmm.fit(d, max_iterations=2000, verbose=False, stop_threshold=1e-9)
```

Comment:
I think because of a reduced roundoff error, fitting seems to be able to continue for longer.